test/Interop/SwiftToCxx/enums/recursive-enum.swift

```diff
@@ -0,0 +1,8 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend %s -module-name Enums -clang-header-expose-decls=all-public -typecheck -verify -emit-clang-header-path %t/enums.h
+
+// RUN: %check-interop-cxx-header-in-clang(%t/enums.h -Wno-unused-private-field -Wno-unused-function -DSWIFT_CXX_INTEROP_HIDE_STL_OVERLAY)
+
+public enum E {
+  case foo([E?])
+}
```

Comment:
Would it make sense to check if the enum is actually emitted into the generated header?